tools/toc.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# Run `markdown-toc` on all Markdown files with tables of contents
+# to see if any of them need an updated TOC.
+#
+# As a side effect, if you run this locally it will generate
+# all needed TOC and update the markdown documents.
+
+let ERROR=0
+
+# Find all markdown files
+for f in $(find . | grep .md); do
+
+	# Only use ones that include a table of contents
+	grep '<!-- tocstop -->' $f > /dev/null
```

Comment:
The way you add a TOC to a page that never had one before is to add `<!-- toc -->` where you want the TOC to go. You don't ever manually write `tocstop`, that gets added automatically on first run by the tool